File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int n = arr.length;

        int l = 1;
        int r = arr[n-1]+k;
        for(int val = l ; val <= r; val++){
            boolean found = false;
            for(int i = 0; i<n; i++){
                if(val == arr[i]){
                    found = true;
                    break;
                }
            }
             if(!found){
                    k--;
                }
                if(k == 0){
                    return val;
                }
           
        }
        return -1;
    }
}

File: 1646-kth-missing-positive-number/1646-kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int n = arr.size();
        int l = 0, h = n-1;
        while(l <= h){
            int mid = (l+h)/2;

            int missingNum = arr[mid]-(mid+1);
            if(missingNum < k){
                l = mid+1;
            }else{
                h = mid-1;
            }
        }
        return l+k;
    }
};